newIDE/app/src/Utils/GDevelopServices/Authentication.js

```diff
@@ -57,22 +57,23 @@ export default class Authentication {
   firebaseUser: ?FirebaseUser = null;
   user: ?Profile = null;
   auth: Auth;
-  _onUserChangeCb: ?() => void = null;
+  _onUserChangeCallBack: ?() => void = null;
 
   constructor() {
     const app = initializeApp(GDevelopFirebaseConfig);
     this.auth = getAuth(app);
     onAuthStateChanged(this.auth, user => {
       if (user) {
         this.firebaseUser = user;
+        if (this._onUserChangeCallBack) this._onUserChangeCallBack();
```

Comment:
I reckon we can even put this at the end of the function, to always call the callback when the user has changed.
This way, we handle the user being disconnected for some reason by firebase (the callback is called => _fetchUserProfile is called => the authenticatedUser is cleared).

Actually what we had before:
![image](https://user-images.githubusercontent.com/1280130/135867024-7510c64b-2442-4695-94e4-c284ba941153.png)